crossvalidation.py

```diff
@@ -80,9 +80,26 @@ def parse_args(args):
     return parser.parse_args()
 
 
-def objective(params, boundary, data, W, results_file=None, tuning_params=None):
+class TrackError:
+    def __init__(self):
+        self.test_error = np.inf
+        self.updated = False
+    def update(self, test_error):
+        if test_error < self.test_error:
+            self.test_error = test_error
+            self.updated = True
+
+def objective(params, boundary, data, W, track_error = None, results_file=None, tuning_params=None):
     if tuning_params is not None:
+        best_params_file = os.path.join(os.path.dirname(results_file), 'best_params.json')
```

Comment:
```suggestion
        best_params_file = "f{results_file}.params.json"
```
This seems less likely to clobber things, and more clear what it's attached to?